### PR TITLE
Logger: Add color support for different log levels

### PR DESCRIPTION
**Pull request by Fijxu**
Looks like this:

image

As I said in the code, this is useful when debugging things in Invidious. Having to find the type of error just by the name of the log level is not comfortable in my opinion.

Feel free to edit anything.

### CODE REVIEW

File: src/invidious/helpers/logger.cr
Comment (syeopite) [inline src/invidious/helpers/logger.cr:15]: Pretty minor nitpick but I'll prefer if you can change `@color` to something that doesn't collide with `#color`. Such as `@use_color`.

It is valid Crystal code but it might cause some confusion to someone starting out with Crystal, or when switching from another language.

File: config/config.example.yml
Comment (syeopite) [inline config/config.example.yml:233]: I think this can just be enabled by default

**Review by syeopite — APPROVED** [review]
LGTM. I'll await some reviews from everyone else to see if the colors should be changed before merging.

**Comment by Fijxu** [comment]
Done, I also added information about the `NO_COLOR` env variable.
Color output also works on docker logs which is pretty neat.

image

**Comment by unixfox** [comment]
I don't think it's a good idea to have this as enabled by default.

I can see the color messing up with some existing log parsing system or something like that.

**Comment by Fijxu** [comment]
> I can see the color messing up with some existing log parsing system or something like that.

Yeah you are right. This could break some automation system that does not support ANSI escape codes. It's better to leave it off by default and it would be great to point it on the wiki that this could be turned on (probably because most people would like to leave this setting on) if you want.

File: src/invidious/helpers/logger.cr
Comment (SamantazFox) [inline src/invidious/helpers/logger.cr:59]: follow-up to my previous comment:
```suggestion
        puts("#{Time.utc} [{{level.id}}] #{message}".colorize(color(LogLevel::{{level.id.capitalize}}))
```

File: config/config.example.yml
Comment (SamantazFox) [inline config/config.example.yml:227]: Also, git reports a trailing white space here!

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
I forgot to say thanks for your contribution :)

**Pull request merged**
